### PR TITLE
fix(platform): Platform types fixes

### LINKED ISSUE #371: Type error in `platform` while running `pnpm run lint`
**Describe the bug**
Lint scripts failing in platform  

**To Reproduce**
1. run `pnpm run lint` in terminal.

**Expected behavior**
Lint should not fail.

**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Version: Latest

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
## Description
 platform types fixes

Fixes #371 

## Dependencies
1. Update the typescript version to 5.3.3
2. Adding `@types/dayjs-precise-range` in platform package.json

## Future Improvements
N/A

## Mentions
@rajdip-b 

## Screenshots of relevant screens
Screenshot from 2024-07-27 11-43-22

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [x] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, dependencies

___

### **Description**
- Added `@types/dayjs-precise-range` to `devDependencies` in `apps/platform/package.json`.
- Updated `typescript` version from `^4.5.3` to `^5.3.3` in `apps/platform/package.json`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Update TypeScript version and add new type definitions</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/package.json

<li>Added <code>@types/dayjs-precise-range</code> to <code>devDependencies</code>.<br> <li> Updated <code>typescript</code> version from <code>^4.5.3</code> to <code>^5.3.3</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/374/files#diff-ccbe95f6e6aa4e3266a5e9cf1bcfee8f323da5b0be41f8cab6e3f8ab98d0ddcd">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by DhairyaMajmudar** [comment]
Shall I also puch the changes done in pnpm-lock file ?

**Comment by rajdip-b** [comment]
Yes! We would need the lockfile changes.

**Review by rajdip-b — APPROVED** [review]

**Comment by rajdip-b** [comment]
@DhairyaMajmudar there's something that I would be glad if you can do. I see that these errors were related to typescript. Now that I see, all the different packages are using their own version of typescript. We don't want that. Could you please move out typescript to the root package.json dev dependency? And delete typescript from all the other package.json files?

**Comment by DhairyaMajmudar** [comment]
> @DhairyaMajmudar there's something that I would be glad if you can do. I see that these errors were related to typescript. Now that I see, all the different packages are using their own version of typescript. We don't want that. Could you please move out typescript to the root package.json dev dependency? And delete typescript from all the other package.json files?

sounds correct, let's do this 🚀

File: apps/api/src/variable/controller/variable.controller.ts
Comment (rajdip-b) [inline apps/api/src/variable/controller/variable.controller.ts:1]: I think you updated the wrong file while merging :/

File: apps/platform/package.json
Comment (rajdip-b) [inline apps/platform/package.json:1]: Remove ts from here aswell

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: apps/api/src/variable/controller/variable.controller.ts
Comment (DhairyaMajmudar) [reply apps/api/src/variable/controller/variable.controller.ts:1]: Yeah removed the changes : )

**Review by rajdip-b — APPROVED** [review]

**Review by rajdip-b — APPROVED** [review]

File: apps/platform/package.json
Comment (DhairyaMajmudar) [reply apps/platform/package.json:1]: Done 👍🏻

**Comment by DhairyaMajmudar** [comment]
First time in life changing 5 package.json files at same time 💀 

Screenshot from 2024-07-27 12-18-41

**Comment by rajdip-b** [comment]
xD

**Pull request merged**